### ISSUE TITLE
Android NavigationPage hides the previous page before back navigation

### ISSUE THREAD

**Issue by leo-mck** [dotnet/maui#21187]
### Description

I am trying to change the animation between pages of a NavigationPage on android. For this I override the 4 xml files

nav_default_enter_anim.xml
nav_default_exit_anim.xml
nav_default_pop_enter_anim.xml
nav_default_pop_exit_anim.xml

but it looks like when going back, the previous page fragment are being destroyed before the animation start. This create an ugly effect, even when using the defaults, altough in that case is hard to notice since the animations run faster. I increased the duration of the animations to be able to see the problem:

https://github.com/dotnet/maui/assets/876986/6f02524a-3414-4ddc-b4ca-4123be884d22

### Link to public reproduction project repository

https://github.com/leo-mck/MauiNavigationPageTestApp

### Version with bug

8.0.7 SR2

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

**Comment by Alex-Dobrynin** [comment]
@Stedy59 have you managed to fix this by custom implementations?

**Comment by yunusefendi52** [comment]
@PureWeen I think it's migraiton issue and a regression from Xamarin Forms to MAUI

**Comment by Stedy59** [comment]
> @Stedy59 have you managed to fix this by custom implementations?

Unfortunately, I have not been able to fix this issue. And, to this day, I have not had any response from either @jsuarezruiz or @PureWeen .

I have a feeling that this is not fixable in the "Maui" code base, but more likely in the Android libs??? @jonathanpeppers

**Comment by jonathanpeppers** [comment]
> I have a feeling that this is not fixable in the "Maui" code base, but more likely in the Android libs???

Do you mean newer AndroidX libraries from Google? That could be a possibility. Generally, the Android workload just provides C# bindings to Java/Kotlin libraries as-is, so behavior in Google's code can change.

I also think the navigation implementation changed quite a bit from Xamarin.Forms to MAUI. 

As for how to create custom animations for navigation, I don't know how to help with this, sorry!

**Comment by Stedy59** [comment]
> > I have a feeling that this is not fixable in the "Maui" code base, but more likely in the Android libs???
> 
> Do you mean newer AndroidX libraries from Google? That could be a possibility. Generally, the Android workload just provides C# bindings to Java/Kotlin libraries as-is, so behavior in Google's code can change.
> 
> I also think the navigation implementation changed quite a bit from Xamarin.Forms to MAUI.
> 
> As for how to create custom animations for navigation, I don't know how to help with this, sorry!

Yes, the implementation TOTALLY changed... but for modal, it still uses the Transaction model.

As for the custom animations, got that working as shown in the vid...

Thank you for responding!!!!

**Comment by rbev** [comment]
It's so depressing to come across these kind of issues, that seem to just have zero care from the MAUI team

Page transitions are one of those things that will make or break your app, and this issue causes a major gulf in the polish between the iOS and Android versions of the app

**Comment by AntoineOrgerit** [comment]
I managed to fix this by adding the `shareInterpolator ` and `interpolator` attributes. See [documentation](https://developer.android.com/guide/fragments/animate#set-transitions) here.

`<set xmlns:android="http://schemas.android.com/apk/res/android"
     android:shareInterpolator="false"
     android:interpolator="@android:anim/decelerate_interpolator">
  <translate android:duration="@android:integer/config_shortAnimTime"
             android:fromXDelta="0"
             android:toXDelta="-100%p"/>
  <alpha android:duration="@android:integer/config_shortAnimTime"
         android:fromAlpha="1"
         android:toAlpha="0" />
</set>`

**Comment by Shalini-Ashokan** [comment]
The issue is not reproduced with the latest Maui NuGet version (9.0.110) on Android platform.

https://github.com/user-attachments/assets/6039dbb3-247a-4950-8421-ca8126c4f07d
